asserts/gpgkeypairmgr.go

```diff
@@ -330,7 +330,28 @@ func (gkm *GPGKeypairManager) findByName(name string) (*gpgKeypairInfo, error) {
 	if err != nil {
 		return nil, err
 	}
-	return nil, fmt.Errorf("cannot find key named %q in GPG keyring", name)
+	return nil, &namedKeyNotFoundError{keyNotFoundError{keyID: name, where: "in GPG keyring"}}
+}
+
+type namedKeyNotFoundError struct {
+	keyNotFoundError
+}
+
+func (e *namedKeyNotFoundError) Error() string {
+	where := ""
+	if e.where != "" {
+		where = fmt.Sprintf(" %s", e.where)
+	}
+	return fmt.Sprintf("cannot find key named %q%s", e.keyID, where)
```

Comment:
nitpick: I would remove this method altogether and reuse that from `keyNotFoundError`. On the other hand, if being able to tell the two errors apart, then I think the texts should be a bit more different. Maybe rename the base class one to `"cannot find key with ID %s"`?